src/components/tx-events.tsx

```diff
@@ -306,27 +317,86 @@ const Item: React.FC<{ event: TransactionEvent; isLast?: boolean }> = ({ event,
   );
 };
 
+export const EventsPaginatedList = ({
+  events,
+}: {
+  events: InfiniteData<ApiResponseWithResultsOffset<TransactionEvent>> | undefined;
+}) => {
+  return events ? (
+    <>
+      {events.pages.map(page => (
+        <Fragment key={page.offset}>
+          {page?.results.map((event, index, arr) => (
+            <Item key={index} event={event} isLast={index === arr.length - 1} />
+          ))}
+        </Fragment>
+      ))}
+    </>
+  ) : null;
+};
+
 const EventsList = ({ events }: { txId: string; events: TransactionEvent[] }) => {
-  // TODO: paginated data
   return (
-    <Box px="loose">
+    <>
       {events.map((event, index, arr) => (
         <Item key={index} event={event} isLast={index === arr.length - 1} />
       ))}
-    </Box>
+    </>
   );
 };
+
 export const Events = ({
   events,
   txId,
+  event_count,
   ...rest
 }: {
-  txId: string;
   events: TransactionEvent[];
+  txId: string;
+  event_count: number;
 } & FlexProps) => {
-  return events?.length ? (
+  const queries = useTransactionQueries();
+  // fetch extra events if there are more than the default returned by the fetch tx api
+  // and display them below
+  const { data, ...actions } = useInfiniteQuery(
```

Comment:
This might be worth extracting to a separate file i.e. `useTxEvents.ts`, the component is pretty large